benchmarking/tpch/answers.py

```diff
@@ -106,12 +106,12 @@ def q4(get_df: GetDFFunc) -> DataFrame:
         (col("O_ORDERDATE") >= datetime.date(1993, 7, 1)) & (col("O_ORDERDATE") < datetime.date(1993, 10, 1))
     )
 
-    lineitems = lineitems.where(col("L_COMMITDATE") < col("L_RECEIPTDATE")).select(col("L_ORDERKEY")).distinct()
+    lineitems = lineitems.where(col("L_COMMITDATE") < col("L_RECEIPTDATE")).limit(1_000_000)
 
     daft_df = (
-        lineitems.join(orders, left_on=col("L_ORDERKEY"), right_on=col("O_ORDERKEY"))
+        orders.join(lineitems, left_on=col("O_ORDERKEY"), right_on=col("L_ORDERKEY"), how="semi")
```

Comment:
Does this not change the join ordering?

Reply:
I don't think so. I'm not sure why it was originally lineitems.join(orders) but either way, it was an inner hash join so whether or not one goes before the other should not make any significant difference in performance.

Reply:
it does change the performance I think, depending on the executor (the new native executor is a bit naive wrt which side it constructs a probe table on for example)

I worry that changing this might affect results for benchmarks that rely on existing behavior

Reply:
I could benchmark it with the original inner join but swapped too. Either way I was not running the new executor for the benchmarks I posted